### PR TITLE
Salary input changes without migrations

### PR DESCRIPTION
**Pull request by cpjmcquillan**
Re-merge of the salary input changes, but without database migrations

### CODE REVIEW

File: app/controllers/hiring_staff/vacancies/pay_package_controller.rb [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+  end
+
+  private
+
+  def pay_package_form_params
+    (params[:pay_package_form] || params).permit(:salary, :benefits)
+  end
+
+  def next_step
+    UploadDocumentsFeature.enabled? ? supporting_documents_school_job_path : candidate_specification_school_job_path
```
Comment (cesidio) [inline app/controllers/hiring_staff/vacancies/pay_package_controller.rb]: You could remove the feature flag here too while you are at it

File: app/form_models/pay_package_form.rb [hunk 2]
```diff
@@ -0,0 +1,7 @@
+class PayPackageForm < VacancyForm
+  include VacancyPayPackageValidations
+
+  def completed?
```
Comment (cesidio) [inline app/form_models/pay_package_form.rb]: This does not make much sense, you can just invoke `valid?`

File: app/controllers/hiring_staff/vacancies/pay_package_controller.rb (hunk 1 above)
Comment (cpjmcquillan) [reply app/controllers/hiring_staff/vacancies/pay_package_controller.rb]: Can I remove the feature flag in a separate PR? It's in a lot of places

File: app/models/concerns/vacancy_pay_package_validations.rb [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+  extend ActiveSupport::Concern
+  include ApplicationHelper
+
+  included do
+    validates :salary, presence: true
+    validates :salary, length: { minimum: 1, maximum: 256 }, if: :salary?
+  end
+
+  def salary=(value)
+    super(sanitize(value, tags: []))
```
Comment (cesidio) [inline app/models/concerns/vacancy_pay_package_validations.rb]: Why are you passing an array with no tags here?

File: app/views/hiring_staff/vacancies/pay_package/show.html.haml [hunk 4]
```diff
@@ -0,0 +1,20 @@
+= render 'hiring_staff/vacancies/vacancy_form_partials/title', form_section_title: 'Pay package', form_section_step: 2, form_section_total_steps: 4
+
+.govuk-grid-row
+  .govuk-grid-column-two-thirds
+    = form_for @pay_package_form, url: school_job_pay_package_path(@vacancy.id), html: { method: "patch" }, builder: GOVUKDesignSystemFormBuilder::FormBuilder do |f|
```
Comment (cesidio) [inline app/views/hiring_staff/vacancies/pay_package/show.html.haml]: Very happy to see the GOVUKDesignSystemFormBuilder 🎊

File: spec/models/vacancy_spec.rb [hunk 5]
```diff
@@ -566,6 +425,14 @@
       expect(vacancy.job_title).to eq(sanitized_title)
     end
 
+    it '#salary' do
```
Comment (cesidio) [inline spec/models/vacancy_spec.rb]: it does not read that well, I would use a context `#salary` and inside would put something like `it strips the tags`

File: app/models/concerns/vacancy_pay_package_validations.rb (hunk 3 above)
Comment (cpjmcquillan) [reply app/models/concerns/vacancy_pay_package_validations.rb]: Because no tags are allowed

**Pull request merged**